lr_face/models.py

```diff
@@ -46,116 +60,158 @@ def predict_proba(self, X: List['FacePair']) -> np.ndarray:
         :return np.ndarray
         """
         scores = []
+        cache_dir = EMBEDDINGS_DIR
         for pair in X:
-            embedding1 = pair.first.get_embedding(
-                self.architecture, output_dir='embeddings')
-            embedding2 = pair.second.get_embedding(
-                self.architecture, output_dir='embeddings')
+            embedding1 = self.embedding_model.embed(pair.first, cache_dir)
+            embedding2 = self.embedding_model.embed(pair.second, cache_dir)
             score = spatial.distance.cosine(embedding1, embedding2)
             scores.append([score, 1 - score])
         return np.asarray(scores)
 
     def __str__(self) -> str:
-        return self.architecture.name
-
-
-class TripletEmbeddingModel(tf.keras.Model):
-    """
-    A subclass of tf.keras.Model that can be used to finetune an existing, pre-
-    trained embedding model using a triplet loss.
+        return f'{self.embedding_model.name}Scorer'
+
+
+class EmbeddingModel:
+    def __init__(self,
+                 base_model: tf.keras.Model,
```

Comment:
is dit neit 'architecture'?

Reply:
Hier heb ik bewust voor `base_model` gekozen; het `EmbeddingModel` gebruikt eigenlijk een onderliggend keras model om de embeddings te bepalen. Het is de `Architecture`'s verantwoordelijkheid om op het moment dat je `Architecture.get_embedding_model()` aanroept, het juiste onderliggende Keras model te laden en dat mee te geven aan het geretourneerde `EmbeddingModel`.

Reply:
dan misschien eerder 'model' of 'keras_model'? base_model impliceert voor mij dat EmbeddingModel er nog iets mee gaat doen ofzo, terwijl dit eigenlijk het volledige model is dat de embeddings oplevert (correct?)

Reply:
Correct

- [ ] Hernoem `base_model` naar `model` of `keras_model`